segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainerTests.java

```diff
@@ -247,6 +251,85 @@ public void testEndToEnd() throws Exception {
         }
     }
 
+    /**
+     * The test creates a segment and then writes some data to it. The method under the test copies the contents of the
+     * segment to a segment with a different name. At the end, it is verified that the new segment has the accurate
+     * contents from the first one.
+     */
+    @Test
+    public void testCopySegment() throws Exception {
+        // Create a storage.
+        @Cleanup
+        val baseStorage =  new InMemoryStorage();
+        @Cleanup
+        val s = new AsyncStorageWrapper(new RollingStorage(baseStorage, new SegmentRollingPolicy(1)), executorService());
+        s.initialize(1);
+        log.info("Created a storage instance");
+
+        String sourceSegmentName = "segment-" + RANDOM.nextInt();
+        String targetSegmentName = "segment-" + RANDOM.nextInt();
+
+        // Create source segment
+        s.create(sourceSegmentName, TIMEOUT).join();
+        val handle = s.openWrite(sourceSegmentName).join();
+
+        // do some writing
```

Comment:
Not sure why we need this in for loop.

Reply:
Removed loop. Writing 10 MB data at once.